libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewSubscriberTest.java

```diff
@@ -0,0 +1,61 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.arch.lifecycle.LifecycleOwner;
+
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class NavigationViewSubscriberTest {
+
+  @Test
+  public void checksRouteObserversAreRemovedWhenUnsubscribe() {
+    NavigationPresenter mockedNavigationPresenter = mock(NavigationPresenter.class);
+    NavigationViewSubscriber theNavigationViewSubscriber = new NavigationViewSubscriber(mockedNavigationPresenter);
+    LifecycleOwner mockedLifecycleOwner = mock(LifecycleOwner.class);
+    NavigationViewModel mockedNavigationViewModel = mock(NavigationViewModel.class, RETURNS_DEEP_STUBS);
```

Comment:
We can move repeated parts (from each test) in @ Before public void setup() method

Reply:
👀 [_Working Effectively with Unit Tests_](https://leanpub.com/wewut/read) 
> when writing tests you should prefer DAMP (Descriptive And Maintainable Procedures) 

That includes a bunch of _procedures_ as [Inline Setup](https://leanpub.com/wewut/read#leanpub-auto-inline-setup) 😅

**1. setUp() method**
- When you remove Setup you may limit the number of times you need to check variables when you first encounter a test. So then you no longer need to worry about irrelevant instances from other tests
- It's a waste of time because you normally need to understand all the Setup each time you have to deal with the test
- Setup advocates argue because if you don't use it you normally create duplication. You can solve it using private methods or builders

I also like to add that normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods. Problem with Setup is that hides the _Arrange_ part completely.

As said, I prefer this way because using DAMP you increase maintainability and readability of the tests. But that's my opinion and we can discuss further 😉 

On a side note, I highly recommend you reading that book (if you haven't already). It's 🔝 💯